### ISSUE TITLE
Fix Tabbing Order - Payment Details

### ISSUE THREAD

**Issue by Zeyber** [SAP/cloud-commerce-spartacus-storefront#4413]
Pressing Tab does not iterate focus through elements correctly according to #4216. Needs fixing so it does.

**Comment by Zeyber** [comment]
Reopening as the tests in #4216 still fail due to links being unreachable by tabbing
